pyomo/gdp/plugins/bound_pretransformation.py

```diff
@@ -164,54 +166,55 @@ def _update_bounds_from_constraints(
             except StopIteration:
                 # There was one but not two: This is what we want.
                 repn = generate_standard_repn(constraint.body)
-                if not repn.is_linear():
+                # If this is a trivial constraint, repn could actually be empty,
+                # so we check that we really do have one linear var now
+                if not repn.is_linear() or len(repn.linear_vars) != 1:
                     continue
                 v = repn.linear_vars[0]
-                v_bounds = self._get_bound_dict_for_var(bound_dict, v)
                 coef = repn.linear_coefs[0]
                 constant = repn.constant
-                self._update_bounds_dict(
-                    v_bounds,
+                lower = (
                     (value(constraint.lower) - constant) / coef
                     if constraint.lower is not None
-                    else None,
+                    else None
+                )
+                upper = (
                     (value(constraint.upper) - constant) / coef
                     if constraint.upper is not None
-                    else None,
-                    bound_dict_key,
+                    else None
+                )
+                if coef < 0:
+                    # we divided by a negative coef above, so flip the constraint
+                    (lower, upper) = (upper, lower)
+                v_bounds = self._get_bound_dict_for_var(bound_dict, v)
+                self._update_bounds_dict(
+                    v_bounds,
+                    lower,
+                    upper,
+                    disjunct if not is_root else None,
                     gdp_forest,
                 )
                 if not is_root:
-                    v_bounds['to_deactivate'].add(constraint)
-
-    def _return_nonNone_bounds(self, parent_lb, parent_ub, lb, ub):
-        if lb is None:
-            # either we replace None with None, or we fill in the parent value
-            # (which we know if the tightest of the ancestral values because
-            # we've already come down the tree once.)
-            lb = parent_lb
-        if ub is None:
-            ub = parent_ub
-        return (lb, ub)
+                    if disjunct in v_bounds['to_deactivate']:
+                        v_bounds['to_deactivate'][disjunct].add(constraint)
+                    else:
+                        v_bounds['to_deactivate'][disjunct] = set([constraint])
```

Comment:
Oh, whoops, good catch. I thought it didn't matter because it was just the order things get deactivated, but that's not true because I have to add constraints in that process sometimes when I'm deactivating the <= side but not the >= side or vice versa.